src/types.ts

```diff
@@ -101,6 +107,17 @@ export type APIQueryHooks<Endpoints extends RoughEndpoints> = {
     options?: RestrictedUseQueryOptions<Endpoints[Route]['Response']>,
   ) => UseQueryResult<Endpoints[Route]['Response']>;
 
+  useAPIInfiniteQuery: <Route extends keyof Endpoints & string>(
+    route: Route,
+    payload: RequestPayloadOf<Endpoints, Route> & {
+      nextPageParamKey: keyof RequestPayloadOf<Endpoints, Route>;
+    },
```

Comment:
It's of course an edge case, but feels worth mentioning that this API does prevent consumers from using an API of their own that accepts a `nextPageParamKey` key.

I had an idea for a small API shift that might let us "not need" that key and still get good type safety -- curious what you think:

```typescript
const query = useAPIInfiniteQuery(
  'GET /items',
  // This is the real "payload"
  { filter: 'some-filter-query-param' },
  {
    // This is a new, custom option.
    // input: the current page response
    // output: a _Partial_ of the request params, to be spread over the input.
    getNextPagePayload: (currentPage: Endpoints['GET /items']['Response']) => {
      return {
        // this key is well-typed and auto-completed because of the Partial
        nextPageToken: parseNextPageToken(currentPage.links.next)
      }
    }
```

If we went this^ way, we'd probably _remove_ the `react-query` `getNextPageParam` option for clarity.

Thoughts?

Reply:
One more (less meaningful) thought: this^ syntax would also allow us to have potentially multiple paging params (though, I'm not sure I've ever seen an API that accepts multiple params)

Reply:
no this is great! I wish I had considered it earlier 🤦 I was worrying I needed to support string `.` for object nesting etc